pandas/core/base.py

```diff
@@ -581,12 +582,17 @@ def _shallow_copy(self, obj, **kwargs):
         """
         return a new object with the replacement attributes
         """
-        if isinstance(obj, self._constructor):
-            obj = obj.obj
-        for attr in self._attributes:
-            if attr not in kwargs:
-                kwargs[attr] = getattr(self, attr)
-        return self._constructor(obj, **kwargs)
+        with warnings.catch_warnings():
```

Comment:
Do you know where this is getting called? Seems strange to have to filter the warning here

Reply:
oh no, we DO not want to do this here at all